components/brave_new_tab_ui/components/search/EngineContext.tsx

```diff
@@ -55,8 +55,9 @@ export function EngineContextProvider(props: React.PropsWithChildren<{}>) {
     // one, or Brave Search.
     // Note: The key for `Google` is the empty string which is falsey so we need
     // to check for undefined here.
-    if (last === undefined || config[last] === undefined) {
-      return Object.keys(config).find(key => config[key] !== undefined) ?? braveSearchHost
+    if (last === undefined || !config[last]) {
+      console.log("Looking for something!", config)
+      return Object.keys(config).find(key => config[key]) ?? braveSearchHost
```

Comment:
Curious we need to return `braveSearchHost` when we can't find any enabled provider.
If user deselect all, we just disable search widget.

Reply:
Hmm, I just tried removing it but we use it as the default search engine if nothing is defined - I think it might be possible to remove but is probably a bigger change.